src/staticarrays_support.jl

```diff
@@ -26,4 +26,12 @@ StructArrays.component(s::StaticArray, i) = getindex(s, i)
     invoke(StructArrays.staticschema, Tuple{Type{<:Any}}, T)
 end
 StructArrays.component(s::FieldArray, i) = invoke(StructArrays.component, Tuple{Any, Any}, s, i)
-StructArrays.createinstance(T::Type{<:FieldArray}, args...) = invoke(createinstance, Tuple{Type{<:Any}, Vararg}, T, args...)
\ No newline at end of file
+StructArrays.createinstance(T::Type{<:FieldArray}, args...) = invoke(createinstance, Tuple{Type{<:Any}, Vararg}, T, args...)
+
+# StaticArrayStyle has no similar defined.
+# Convert to `DefaultArrayStyle` to return a sized (Struct)Array.
+# TODO: return a StaticArray?
```

Comment:
Yeah, I think it'd be best to return a proper `StaticArray` (same as you'd get with `SVector(1, 2) .+ SVector(1, 2)`) and to test we are not allocating, ie something like

```julia
julia> function f()
           s = StructArray{ComplexF64}((SVector(1., 2., 3.), SVector(0., 0., 0.)))
           return broadcast(log, s)
       end

julia> @allocated f()
```

should ideally return `0`.

Reply:
With [6796aae9](https://github.com/N5N3/StaticArrays.jl/commit/6796aae90dd8a274d4c12c5e6bbaa60dac2ee324) for `StaticArrays`, the goal can be done. We just need to define
```julia
function Base.copy(bc::Broadcasted{StructArrayStyle{StaticArrayStyle{N},N}}) where {N}
    bc′ = convert(Broadcasted{StaticArrayStyle{N}}, bc)
    return copy(bc′)
end
Size(::Type{SA}) where {SA<:StructArray} = Size(fieldtype(fieldtype(SA, 1), 1))
isstatic(a::SA) where {SA<:StructArray} = cst(SA) isa StaticArrayStyle
function similar_type(::Type{SA}, ::Type{T}, s::Size{S}) where {SA<:StructArray, T, S}
    return similar_type(fieldtype(fieldtype(SA, 1), 1), T, s)
end
```
Now, we have
```julia
julia> a = @SArray [1,2]
2-element SVector{2, Int64} with indices SOneTo(2):
 1
 2

julia> b = StructArray{Complex{Int}}((a,a))
2-element StructArray(::SVector{2, Int64}, ::SVector{2, Int64}) with eltype Complex{Int64} with indices SOneTo(2):
 1 + 1im
 2 + 2im

julia> @btime $a .+ $b
  2.500 ns (0 allocations: 0 bytes)
2-element SVector{2, Complex{Int64}} with indices SOneTo(2):
 2 + 1im
 4 + 2im

julia> @btime $b .+ $b
  2.000 ns (0 allocations: 0 bytes)
2-element SVector{2, Complex{Int64}} with indices SOneTo(2):
 2 + 2im
 4 + 4im
```